modules/javafx.base/src/main/java/javafx/collections/FXCollections.java

```diff
@@ -391,6 +397,33 @@ public static <E> ObservableList<E> concat(ObservableList<E>... lists) {
         return observableList(backingList);
     }
 
+    /**
+     * Returns an unmodifiable {@code ObservableList} that contains the concatenation of the contents
+     * of the specified observable lists. The returned list is a view onto the source lists, which means
+     * that read operations are delegated to the appropriate source list, and attempts to modify the
+     * returned list result in an {@link UnsupportedOperationException}.
+     * <p>
+     * In contrast to {@link #concat(ObservableList[])}, the list returned from this method is updated
+     * when any of the source lists are changed.
+     *
+     * @param lists the source lists
+     * @param <E> the element type
+     * @return an unmodifiable {@code ObservableList} view that contains the concatenation of the source lists
+     * @throws IndexOutOfBoundsException if number of elements contained in all source lists
+     *                                   exceeds {@link Integer#MAX_VALUE}
+     * @since 21
```

Comment:
This method could use some specification about how it handles nulls:

- NPE when provided with `concatenatedObservableList((ObservableList<E>[]) null)`
- NPE when any element is `null`